multiqc/modules/multivcfanalyzer/multivcfanalyzer.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from MultiVCFAnalyzer """
+
+from __future__ import print_function
+from collections import OrderedDict
+import logging
+import json
+
+from multiqc.plots import bargraph
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+class MultiqcModule(BaseMultiqcModule):
+    """ MultiVCFAnalyzer module """
+
+    def __init__(self):
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(name='MultiVCFAnalyzer', anchor='multivcfanalyzer',
+        href="https://github.com/alexherbig/MultiVCFAnalyzer",
+        info="""MultiVCFAnalyzer reads multiple VCF files as produced by the GATK UnifiedGenotyper and after filtering provides the combined genotype calls in a number of formats that are suitable for follow-up analyses such as phylogenetic reconstruction, SNP effect analyses, population genetic analyses etc.""")
+
+        # Find and load any MultiVCFAnalyzer reports
+        self.mvcf_data = dict()
+
+        # Find and load JSON file
+        for f in self.find_log_files('multivcfanalyzer', filehandles=True):
+            self.parse_data(f)
+
+        # Filter samples
+        self.mvcf_data = self.ignore_samples(self.mvcf_data)
+
+        # Return if no samples found
+        if len(self.mvcf_data) == 0:
+            raise UserWarning
+
+        # Add in extra columns to data file
+        self.compute_perc_hets()
+
+        # Save data output file
+        self.write_data_file(self.mvcf_data, 'multiqc_multivcfanalyzer_metrics')
+
+        # Add to General Statistics
+        self.addSummaryMetrics()
+
+
+    def parse_data(self, f):
+        try:
+            data = json.load(f['f'])
+        except Exception as e:
+            log.debug(e)
+            log.warn("Could not parse MultiVCFAnalyzer JSON: '{}'".format(f['fn']))
+            return
+
+        # Parse JSON data to a dict
+        for s_name in data:
```

Comment:
```suggestion
        for s_name, metrics in data.get('metrics', {}).items():
```
This negates the need for the `if (s_name == 'metrics'):` later on, and means that if the JSON doesn't have a `metrics` key for whatever reason it'll get an empty dict instead and just not do the loop.

Reply:
Ok, I think you can refactor this whole parsing block as follows:

```python
for s_name, metrics in data.get('metrics', {}).items():
    s_name = self.clean_s_name(s_name, f['root'])
    if s_clean in self.mvcf_data:
        log.debug("Duplicate sample name found! Overwriting: {}".format(s_clean))
    self.add_data_source(f, s_name)
    self.mvcf_data[s_clean] = metrics
```